### PR TITLE
[ADD] [11.0] event_survey: create module

### PR DESCRIPTION
**Pull request by i-vyshnevska**
On click  smartbutton "Survey" from event.event, we able to chose from a list of survey templates in order not to recreate a new survey from scratch.
[BSFER-251](https://jira.camptocamp.com/browse/BSFER-251)

### CODE REVIEW

File: event_survey/README.rst
Comment (simahawk) [inline event_survey/README.rst]: RST broken.
If you move to readme folder structure https://github.com/OCA/maintainer-tools/tree/master/template/module/readme you can forget about this issues, at least for titles ;)

File: event_survey/__init__.py
Comment (simahawk) [inline event_survey/__init__.py]: copy not needed on `__init__.py`

File: event_survey/__manifest__.py
Comment (simahawk) [inline event_survey/__manifest__.py]: :fire:

File: event_survey/__manifest__.py
Comment (simahawk) [inline event_survey/__manifest__.py]: https://github.com/camptocamp/odoo-event-addons

File: event_survey/models/event_event.py
Comment (simahawk) [inline event_survey/models/event_event.py]: copy=False

File: event_survey/models/event_survey.py
Comment (simahawk) [inline event_survey/models/event_survey.py:10]: copy=False

File: event_survey/models/event_survey.py
Comment (simahawk) [inline event_survey/models/event_survey.py]: *Survey

File: event_survey/wizard/event_survey_wizard.py
Comment (simahawk) [inline event_survey/wizard/event_survey_wizard.py]: do you really need the wrapping div too? The link w/ `data-oe-*` attrs should be enough.

File: event_survey/wizard/event_survey_wizard.py
Comment (simahawk) [inline event_survey/wizard/event_survey_wizard.py]: is_template can be skipped given that copy=False is applied

File: event_survey/wizard/event_survey_wizard.py
Comment (simahawk) [inline event_survey/wizard/event_survey_wizard.py]: I know odoo code is plenty of examples like this as they do not care that much about HTML, still, ` data-oe-model=survey.survey` should be ` data-oe-model="survey.survey"`

File: event_survey/wizard/event_survey_wizard_view.xml
Comment (simahawk) [inline event_survey/wizard/event_survey_wizard_view.xml]: I don't think we should set new surveys as templates by default as we are creating them from a template already.

File: event_survey/views/event_event.xml
Comment (simahawk) [inline event_survey/views/event_event.xml]: no spaces around `=`

**Review by simahawk — CHANGES_REQUESTED** [review]

File: event_survey/wizard/event_survey_wizard_view.xml
Comment (i-vyshnevska) [reply event_survey/wizard/event_survey_wizard_view.xml]: this context affect  on survey which you create from wizard as template, and use it as base to create new  survey for event, if we have domain here only template it's logical that survey created from here is_template

File: event_survey/wizard/event_survey_wizard.py
Comment (simahawk) [inline event_survey/wizard/event_survey_wizard.py]: too many parenthesis... a list of tuple is enough ;)

**Review by simahawk — COMMENTED** [review]
small issue, LG

File: event_survey/wizard/event_survey_wizard.py
Comment (grindtildeath) [inline event_survey/wizard/event_survey_wizard.py]: My 2 cents, but IMO would make sense to name the wizard with what it does instead of just calling it 'Wizard', e.g `event.survey.generator` gives a better idea of what it does without having to check the code or try it out. (and it's also closer to the name of its action https://github.com/camptocamp/odoo-event-addons/pull/24/files#diff-30bfa33a8eb2f231d2a8a66f8f3b7621R25) :wink: 

Moreover, keeping `event.survey.*` as prefix of your model name, and adding the rest after keeps it in line with odoo naming convention and OCA guidelines.

File: event_survey/readme/DESCRIPTION.rst
Comment (grindtildeath) [inline event_survey/readme/DESCRIPTION.rst]:
```suggestion
This module extends the functionality of event and survey modules. It allows
```

File: event_survey/readme/DESCRIPTION.rst
Comment (grindtildeath) [inline event_survey/readme/DESCRIPTION.rst]: Some rephrasing would be nice here.

Comment (grindtildeath) [inline event_survey/readme/DESCRIPTION.rst]:
```suggestion
users to mark an existing survey as a survey template, so it can be reused to generate new surveys on events according to the template, without the need to create the survey from scratch.```

File: event_survey/wizard/event_survey_wizard.py
Comment (grindtildeath) [inline event_survey/wizard/event_survey_wizard.py]:
```suggestion
        required=True,
```
:warning:

**Review by grindtildeath — COMMENTED** [review]
Some nitpicking and a dangerous typo. 

(I reviewed as comment only not to block the PR while I'm away for one more week)

File: event_survey/wizard/event_survey_wizard.py
Comment (i-vyshnevska) [inline event_survey/wizard/event_survey_wizard.py]: @simahawk 
title - translated field, if user works under other language then English he would see 'template(copy)', if we need it, must do it manually

Comment (simahawk) [reply event_survey/wizard/event_survey_wizard.py]: no problem, they use only FR

File: event_survey/wizard/event_survey_wizard.py
Comment (simahawk) [reply event_survey/wizard/event_survey_wizard.py]: I agree w/ @grindtildeath.  If you want to keep something that mentions clearly this is a wizard you can append `wiz` to it, like `event.survey.generator.wiz`

File: event_survey/tests/test_survey.py
Comment (simahawk) [inline event_survey/tests/test_survey.py]: you've just created it and used it.... I bet is True ;)

File: event_survey/tests/test_survey.py
Comment (simahawk) [inline event_survey/tests/test_survey.py]: they are related via o2m, I bet they are not the same ;)

File: event_survey/views/event_event.xml
Comment (simahawk) [inline event_survey/views/event_event.xml:10]: bad indent

**Review by simahawk — COMMENTED** [review]
few small details but overall looks good

File: event_survey/i18n/fr.po
Comment (sebalix) [inline event_survey/i18n/fr.po]: "Generate"

File: event_survey/wizard/event_survey_generator_view.xml
Comment (sebalix) [inline event_survey/wizard/event_survey_generator_view.xml]: Generate

File: event_survey/wizard/event_survey_generator_view.xml
Comment (sebalix) [inline event_survey/wizard/event_survey_generator_view.xml:10]: Not mandatory, can we add `event_id` in readonly here? With these kind of wizards I like to display the record we are going to process as the pop-up hide the current form. If the readonly makes trouble (maybe the value won't be sent to the controller), we can now use the new `force_save: True` context key :)

**Review by sebalix — CHANGES_REQUESTED** [review]

File: event_survey/wizard/event_survey_generator_view.xml
Comment (i-vyshnevska) [reply event_survey/wizard/event_survey_generator_view.xml:10]: don't think so, works properly, use key just to try :) 
and also can create template if we need

Comment (sebalix) [reply event_survey/wizard/event_survey_generator_view.xml:10]: I was speaking about adding a new `<field name="event_id"/>` but it was only a suggestion, not blocking.

File: event_survey/views/event_event.xml
Comment (i-vyshnevska) [reply event_survey/views/event_event.xml:10]: I updated this,  but it's not showing us outdated, or I didn't understand where exactly the problem

**Review by simahawk — APPROVED** [review]

**Pull request merged**
